src/libos/crates/io-uring-callback/src/lib.rs

```diff
@@ -375,17 +391,42 @@ impl IoUring {
     ///
     /// Upon receiving completed I/O, the corresponding user-registered callback functions
     /// will get invoked and the `IoHandle` (as a `Future`) will become ready.
-    pub fn poll_completions(&self) {
+    /// Return the number of I/O completions. The Completions of Cancel do not count in the number.
```

Comment:
Well, I think it is time to clear things up. We have three concepts that may confuse ourselves as well as users.

*  I/O io_uring completions and completion queues. The completion in this context means the request is handled by the kernel, and the request could succeed, fail, or get cancelled.
* The completion state of `IoHandle`, which is `IoState::Completed`. 
*  The completions of cancel operations, which is `IoState::Cancelled`.

The name of `poll_completions` is originally meant to use the first meaning, but the comments of "completions of cancel do not count" effectively shifts the meaning to the third one. To make the matter worse, the internal implementation and the users of `IoHandle::state` method would take the second meaning. Admittedly, this situation is not good: confused concepts lead to implementation bugs.

To address this confusion, I make the following proposal.

* Let's only talk about _completions_ in the original meaning used by io_uring literatures. That is, completion means a request is handled by the kernel, and the request could succeed, fail, or get cancelled.
* Rename `IoState::Completed` to `IoState::Processed` (any better name suggestion?).
* When referring the callback function argument, always stick with the name `completion_callback`, to emphasize the callback is called when the io_uring request is completed, including the case of being cancelled.
* Fix the implementation of `poll_completions` to count cancellations and fix the docs.
* Fix other docs or add more docs accordingly.